src/ray/raylet/scheduling/policy/bundle_scheduling_policy.cc

```diff
@@ -441,10 +441,26 @@ SchedulingResult BundleStrictPackSchedulingPolicy::Schedule(
     return SchedulingResult::Infeasible();
   }
 
-  auto best_node = GetBestNode(aggregated_resource_request,
-                               candidate_nodes,
-                               options,
-                               available_cpus_before_bundle_scheduling);
+  std::pair<scheduling::NodeID, const Node *> best_node(scheduling::NodeID::Nil(),
+                                                        nullptr);
+  if (!options.bundle_strict_pack_soft_target_node_id.IsNil()) {
+    if (candidate_nodes.contains(options.bundle_strict_pack_soft_target_node_id)) {
+      best_node = GetBestNode(
```

Comment:
why don't we just support get_preferred_node here? 

```
scheduling::NodeID HybridSchedulingPolicy::GetBestNode(
    std::vector<std::pair<scheduling::NodeID, float>> &node_scores,
    size_t num_candidate_nodes,
    std::optional<scheduling::NodeID> preferred_node_id,
    float preferred_node_score) const {
```

Looks like the hybrid policy already supports it.